src/prng.c

```diff
@@ -0,0 +1,92 @@
+/******************************************************************************
+ *
+ * Copyright 2017 Xaptum, Inc.
+ * 
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ * 
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License
+ *
+ *****************************************************************************/
+
+#include <ecdaa/prng.h>
+
+#ifndef DISABLE_LIBSODIUM_RNG_SEED_FUNCTION
+
+#include <sodium.h>
+
+#if defined(__linux__)
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <linux/random.h>
+#endif  // defined(__linux__)
+
+int ecdaa_prng_init(struct ecdaa_prng *prng_in)
+{
+    int ret = 0;
+
+    prng_in->initialized = ECDAA_PRNG_INITIALIZED_NO;
+
+    do {
+        // Note: We don't have to worry about the race-condition here.
+        // `sodium_init` can be called multiple times, and from multiple threads.
+        if (-1 == sodium_init()) {
+            ret = -1;
+            break;
+        }
+#if defined(__linux__) && defined(RNDGETENTCNT)
```

Comment:
Move this block to a separate function.

Reply:
This check should also go before the call to `sodium_init`. Otherwise the entropy might have been low when `sodium_init` was run, but high when the check is being made.

Reply:
:+1: